docs/reference/filters.md

```diff
@@ -1084,6 +1084,99 @@ forwardToken("X-Tokeninfo-Forward")
 forwardToken("X-Tokeninfo-Forward", "access_token")
 ```
 
+## oauthGrant
+
+Enables authentication and authorization with an OAuth2 authorization code grant flow as
+specified by [RFC 6749 Section 1.3.1](https://tools.ietf.org/html/rfc6749#section-1.3.1).
+Automatically redirects unauthenticated users to log in at their provider's authorization
+endpoint. Supports token refreshing and stores access and refresh tokens in an encrypted
+cookie. Supports credential rotation for the OAuth2 client ID and secret.
+
+The filter consumes and drops the grant token request cookie to prevent it from leaking
+to untrusted downstream services.
+
+The filter must be used in conjuction with the [grantCallback](#grantCallback) filter
+where the OAuth2 provider can redirect authenticated users with an authorization code.
+
+The filter may be used with the [grantClaimsQuery](#grantClaimsQuery) filter to perform
+authz and access control.
+
+See the [tutorial](../tutorials/auth.md#oauth2-authorization-grant-flow) for step-by-step 
+instructions.
+
+Examples:
+
+```
+callback:
+    Path("/oauth/callback")
+    -> grantCallback()
+    -> <shunt>;
+
+all:
+    *
+    -> oauthGrant()
```

Comment:
this route is added by the PreProcessor

Reply:
I think you meant to refer to the `grantCallback` route that was also registered here. I removed it from the documentation as it is not directly related to `oauthGrant`

Reply:
yes